test/labeling/model/test_label_model.py

```diff
@@ -117,7 +117,8 @@ def test_generate_O(self):
 
         # Higher order returns same matrix (num source = num cliques)
         # Need to test c_tree form
-        label_model._generate_O(L + 1, higher_order=True)
+        label_model.higher_order = True
```

Comment:
we shouldn't have `higher_order = True` tests in the independent label model version, just to maintain the difference

Reply:
also there should be a warning/error when `higher order=True` but the independent LabelModel is being called

Reply:
The label model already has some higher order code (it had some even before this PR, but now it has more). e.g. The function to create l_aug. We should have some tests to improve coverage for that right?

Reply:
got it, this can stay as is then